utils/network.py

```diff
@@ -0,0 +1,255 @@
+import networkx as nx
+import pandas as pd
+import plotly.graph_objects as go
+
+
+def name_identifier(uuid: str, dfs: list[pd.DataFrame]) -> str:
+    """Returns the name of the entity given the entity's uuid
+
+    Args:
+        uuid: the uuid of the entity
+        List of dfs: dataframes that have a uuid column, and an 'name' or
+        'full_name' column
+    Return:
+        The entity's name
+    """
+    for df in dfs:
+        # first, check orgs df:
+        if "name" in df.columns:
+            name_in_org = df.loc[df["id"] == uuid]
+            if len(name_in_org) > 0:
+                return name_in_org.iloc[0]["name"]
+        # theoretically it must be in inds if not in orgs, but for the sample
+        # data this might not be the case
+
+        if "full_name" in df.columns:
+            name_in_ind = df.loc[df["id"] == uuid]
+            if len(name_in_ind) > 0:
+                return name_in_ind.iloc[0]["full_name"]
+    return None
+
+
+def combine_datasets_for_network_graph(dfs: list[pd.DataFrame]) -> pd.DataFrame:
+    """Combines the 3 dataframes into a single dataframe to create the graph
+
+    Given the inds, orgs, and transactions dataframes, the func first finds the
+    recipient_id in the transaction dataframe in either the org or inds
+    dataframes and adds the name of the recipient to the transaction df. Then,
+    the inds and orgs dfs are merged with the transaction df and concatenated
+    with the contributions amount aggregated, making a final dataframe of the
+    merged transactions and entity dataframes.
+
+    Args:
+        list of dataframes in the order: [inds_df, orgs_df, transactions_df]
+        Transactions dataframe with at least column: 'recipient_id'
+        Individuals dataframe with at least column: 'full_name'
+        Organizations dataframe with at least column: 'name'
+
+    Returns
+        A merged dataframe with aggregate contribution amounts between entitites
+    """
+
+    inds_df, orgs_df, transactions_df = dfs
+
+    # first update the transactions df to have a recipient name tied to id
+    transactions_df["recipient_name"] = transactions_df["recipient_id"].apply(
+        name_identifier, args=([orgs_df, inds_df],)
+    )
+
+    # next, merge the inds_df and orgs_df with the transactions_df
+    inds_trans_df = pd.merge(
+        inds_df, transactions_df, how="left", left_on="id", right_on="donor_id"
+    )
+    inds_trans_df = inds_trans_df.dropna(subset=["amount"])
+    orgs_trans_df = pd.merge(
+        orgs_df, transactions_df, how="left", left_on="id", right_on="donor_id"
+    )
+    orgs_trans_df = orgs_trans_df.dropna(subset=["amount"])
+    orgs_trans_df = orgs_trans_df.rename(columns={"name": "full_name"})
```

Comment:
I think it makes more sense to change the column name from 'full_name' to 'name' in the ind table but this works -- if you do this step ahead of time, you can reduce the checks in the name identifier function

Reply:
@averyschoen  I had thought of that, but applying the name_identifier takes a long time to run. If I change the name before calling it, then the apply function is more likely to query both dataframes since the 'name' column will be in both, which takes longer than specifying to only check the one guaranteed dataframe

Reply:
gotcha -- still need you to fix the comments in the name identifier function